openedx/core/lib/api/authentication.py

```diff
@@ -125,3 +127,110 @@ def _get_dot_token(self, access_token):
         """
         token_query = dot_models.AccessToken.objects.select_related('user')
         return token_query.filter(token=access_token).first()
+
+
+class OAuth2Authentication(BaseAuthentication):
+    """
+    This is created to be a drop in replacement for django-rest-framework-oauth oauth2Authentication class.
+    This is based on NOAuth2AuthenticationAllowINactiveUsers
+    """
+    www_authenticate_realm = 'api'
+    allow_query_params_token = settings.DEBUG
+
+    # currently, active users are users that confirm their email. In case, we want to allow users to 
+    # access edx without confirming(in case of some mobile users) their email, overwrite this by setting
+    # it to True
+    allow_inactive_users = False
+
+    def authenticate(self, request):
+        """
+        Returns two-tuple of (user, token) if access token authentication
+        succeeds, raises an AuthenticationFailed (HTTP 401) if authentication
+        fails or None if the user did not try to authenticate using an access
+        token.
+        """
+
+        set_custom_metric("OAuth2Authentication", "Failed")
+        auth = get_authorization_header(request).split()
+
+        if len(auth) == 1:
+            raise AuthenticationFailed({
```

Comment:
I think these failure responses are changing from String message to JSON.  Might be ok to make this change, but we should document it the commit, and it adds some risk of any individual endpoint causing an issue during failures.